pandas/tests/arrays/test_boolean.py

```diff
@@ -391,13 +391,101 @@ def test_scalar(self, data, all_logical_operators):
 
     def test_array(self, data, all_logical_operators):
         op_name = all_logical_operators
+        if "or" in op_name:
+            pytest.skip("confusing")
         other = pd.array([True] * len(data), dtype="boolean")
         self._compare_other(data, op_name, other)
         other = np.array([True] * len(data))
         self._compare_other(data, op_name, other)
         other = pd.Series([True] * len(data), dtype="boolean")
         self._compare_other(data, op_name, other)
 
+    def test_kleene_or(self):
+        # A clear test of behavior.
+        a = pd.array([True] * 3 + [False] * 3 + [None] * 3, dtype="boolean")
+        b = pd.array([True, False, None] * 3, dtype="boolean")
+        result = a | b
+        expected = pd.array(
+            [True, True, True, True, False, None, True, None, None], dtype="boolean"
+        )
+        tm.assert_extension_array_equal(result, expected)
+
+        result = b | a
+        tm.assert_extension_array_equal(result, expected)
+
+    def test_kleene_or_scalar(self):
+        a = pd.array([True, False, None], dtype="boolean")
+        result = a | np.nan  # TODO: pd.NA
+        expected = pd.array([True, None, None], dtype="boolean")
+        tm.assert_extension_array_equal(result, expected)
+
+        result = np.nan | a  # TODO: pd.NA
+        tm.assert_extension_array_equal(result, expected)
+
```

Comment:
Can you also test the case where the scalar is True or False here? (that is still related to Kleene, since there is a missing value in the array)

Reply:
Or integrate with test case below?